.github/workflows/native-tests.yml

```diff
@@ -39,21 +45,40 @@ jobs:
         run: mvn -B verify -Djedismock.version=${{ env.jm_version }}
         working-directory: native_tests/jm-test-server
 
-      - name: Run jm-test-server in background
-        run: java -jar target/jm-test-server-1.0-SNAPSHOT-jar-with-dependencies.jar &
-        working-directory: native_tests/jm-test-server
-
       - name: Prepare TCL test run
         run: |
           sudo chmod +x ./gen-test-certs.sh
           sudo chmod +x ./runtest
           ./gen-test-certs.sh
         working-directory: native_tests/linux
 
-      - name: Run native tests
+      - name: incr
         run: |
+          java -jar jm-test-server/target/jm-test-server-1.0-SNAPSHOT-jar-with-dependencies.jar &
+          PID=$!
+          cd linux
+          wait4x redis redis://127.0.0.1:39807
           ./runtest --host 127.0.0.1 --port 39807 --single unit/type/incr
+          kill $PID
+        working-directory: native_tests
+
+      - name: hash
+        run: |
+          java -jar jm-test-server/target/jm-test-server-1.0-SNAPSHOT-jar-with-dependencies.jar &
+          PID=$!
+          cd linux
+          wait4x redis redis://127.0.0.1:39807
           ./runtest --host 127.0.0.1 --port 39807 --single unit/type/hash
+          kill $PID
+        working-directory: native_tests
+
+      - name: list
+        run: |
+          java -jar jm-test-server/target/jm-test-server-1.0-SNAPSHOT-jar-with-dependencies.jar &
+          PID=$!
+          cd linux
+          wait4x redis redis://127.0.0.1:39807
           ./runtest --host 127.0.0.1 --port 39807 --single unit/type/list-2
           ./runtest --host 127.0.0.1 --port 39807 --single unit/type/list-3

```

Comment:
it is also possible to include third (unit/type/list) test set by commenting out three tests: `Linked BRPOPLPUSH`, `Circular BRPOPLPUSH` and `Check if list is still ok after a DEBUG RELOAD -`